fabtests/functional/shared_ctx.c

```diff
@@ -67,20 +67,22 @@ static int get_dupinfo(void)
 {
 	struct fi_info *hints_dup;
 	int ret;
+	char *src_addr;
+	char *dest_addr;
 
-       /* Get a fi_info corresponding to a wild card port. The first endpoint
-	* should use default/given port since that is what is known to both
-	* client and server. For other endpoints we should use addresses with
-	* random ports to avoid collision. fi_getinfo should return a random
-	* port if we don't specify it in the service arg or the hints. This
-	* is used only for non-MSG endpoints. */
+	/* Get a fi_info corresponding to a wild card port. The first endpoint
+	 * should use default/given port since that is what is known to both
+	 * client and server. For other endpoints we should use addresses with
+	 * random ports to avoid collision. fi_getinfo should return a random
+	 * port if we don't specify it in the service arg or the hints. This
+	 * is used only for non-MSG endpoints. */
 
 	hints_dup = fi_dupinfo(hints);
 	if (!hints_dup)
 		return -FI_ENOMEM;
 
-	free(hints_dup->src_addr);
-	free(hints_dup->dest_addr);
+	src_addr = hints_dup->src_addr;
+	dest_addr = hints_dup->dest_addr;
```

Comment:
This change should be moved to a separate patch, so that the reason for the change can be captured.  Should also add a  comment here referencing the problem (pointing to the more descriptive comment about the issue).